lib/utils/funding.js

```diff
@@ -0,0 +1,145 @@
+'use strict'
+
+const URL = require('url').URL
+
+exports.getFundingInfo = getFundingInfo
+exports.validFundingUrl = validFundingUrl
+
+// Is the value of a `funding` property of a `package.json`
+// a valid type+url for `npm fund` to display?
+function validFundingUrl (funding) {
+  if (!funding || !funding.url) {
+    return false
+  }
+
+  try {
+    var parsed = new URL(funding.url)
+  } catch (error) {
+    return false
+  }
+
+  if (
+    parsed.protocol !== 'https:' &&
+    parsed.protocol !== 'http:'
+  ) return false
+
+  return Boolean(parsed.host)
+}
+
+function getFundingInfo (idealTree, opts) {
+  let length = 0
+  const seen = new Set()
+  const { countOnly } = opts || {}
+  const empty = () => Object.create(null)
+  const _trailingDependencies = Symbol('trailingDependencies')
+
+  function tracked (name, version) {
+    const key = String(name) + String(version)
+    if (seen.has(key)) {
+      return true
+    }
+    seen.add(key)
+  }
+
+  function retrieveDependencies (dependencies) {
+    const trailing = dependencies[_trailingDependencies]
+
+    if (trailing) {
+      return Object.assign(
+        empty(),
+        dependencies,
+        trailing
+      )
+    }
+
+    return dependencies
+  }
+
+  function hasDependencies (dependencies) {
+    return dependencies && (
+      Object.keys(dependencies).length ||
+      dependencies[_trailingDependencies]
+    )
+  }
+
+  function getFundingDependencies (tree) {
+    const deps = tree && tree.dependencies
+    if (!deps) return empty()
+
+    // broken into two steps to make sure items appearance
+    // within top levels takes precedence over nested ones
+    return (Object.keys(deps)).map((key) => {
+      const dep = deps[key]
+      const { name, funding, version } = dep
```

Comment:
```suggestion
      const { name, version } = dep
      const funding = dep.funding || dep.collective;
```

There are many packages using the `collective` property today to expose their Open Collective URL. Would be great to support that when `funding` is not found, while we wait for them to add it. I

The syntax of `collective` is exactly the same as `funding`, making the implementation uncomplicated.

It's also nice that this allow anyone to test the feature and see results today, while in the current state it's giving me only empty results for all the projects I tested.